File: src/two_sum.cpp
#include "main.hpp"
#include <map>
#include <utility>

std::vector<int> twoSum(std::vector<int>& nums, int target) {
	std::map<int, int> map;
	for(int i = 0; i < nums.size(); i++) {
		int difference = target - nums[i];
		if(map.find(difference) != map.end()) {
			return {map[difference], i};
		}
		map.insert(std::make_pair(nums[i], i));
	}
	return {};
}

File: problems/arrays-and-hashing/two_sum.py
def twoSum(nums, target):
    dict = {}
    
    for (i, x) in enumerate(nums):
        if x in dict.keys():
            return [dict[x], i]
        else:
            complement = target - x
            dict[complement] = i
            
    return []

print(twoSum(nums = [3,4,5,6], target = 7))